certification/internal/engine/openshift.go

```diff
@@ -15,13 +16,26 @@ import (
 	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/cli"
 	client "github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/client"
+	"k8s.io/client-go/tools/clientcmd"
 )
 
-type OpenshiftEngine struct {
+type openshiftEngine struct {
 	KubeConfig *rest.Config
 }
 
-func (pe OpenshiftEngine) CreateNamespace(name string, opts cli.OpenshiftOptions) (*cli.OpenshiftReport, error) {
+func NewOpenshiftEngine() *cli.OpenshiftEngine {
+	k8sconfig, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
```

Comment:
Should we be taking into account the possibility of running within a cluster? Doesn't the BuildConfig change? Maybe attempt to use the in-cluster first, then fall back to this?

Reply:
The in-cluster case assumes the cluster under test is the same cluster where preflight runs. I think we've historically discouraged this (because we can't guarantee the two processes won't cross paths... they shouldn't, but well).

Reply:
That's fair.